node/src/accountant/scanners/mod.rs

```diff
@@ -894,10 +897,75 @@ impl ReceivableScanner {
             earning_wallet,
             receivable_dao,
             banned_dao,
+            persistent_configuration,
             financial_statistics,
         }
     }
 
+    fn update_start_block_while_no_received_transactions_found(
+        &mut self,
+        new_start_block: u64,
+        logger: &Logger,
+    ) {
+        info!(
+            logger,
+            "No new received payments were detected during the scanning process."
+        );
+        match self
+            .persistent_configuration
+            .set_start_block(new_start_block)
+        {
+            Ok(()) => debug!(logger, "Start block updated to {}", new_start_block),
+            Err(e) => panic!(
+                "Attempt to set new start block to {} failed due to: {:?}",
+                new_start_block, e
+            ),
+        }
+    }
+
+    fn update_account_balances_start_block_and_stats(
+        &mut self,
+        msg: &ReceivedPayments,
+        logger: &Logger,
+    ) {
+        let mut txn = self
+            .receivable_dao
+            .as_mut()
+            .more_money_received(msg.timestamp, &msg.payments);
+
+        let new_start_block = msg.new_start_block;
+        match self
```

Comment:
This code repetition can be avoided if we write functions as mentioned in a comment few comments above this one.

Reply:
I told you up there that is not repetition.